src/markdown.cpp

```diff
@@ -1856,15 +1937,28 @@ static QCString extractTitleId(QCString &title, int level)
   {
     std::string id = match[1].str();
     title = title.left(match.position());
+    int &count = id_count[id];
+    if (count > 0) {
+      warn(m_fileName,
+           m_lineNr,
+           "A automatically generated id already has the name '%s'!",
+           id.c_str());
+    }
     //printf("found match id='%s' title=%s\n",id.c_str(),qPrint(title));
     AUTO_TRACE_EXIT("id={}",id);
-    return QCString(id);
+    return id;
   }
-  if ((level > 0) && (level <= Config_getInt(TOC_INCLUDE_HEADINGS)))
-  {
-    static AtomicInt autoId { 0 };
+  if ((level > 0) && (level <= Config_getInt(TOC_INCLUDE_HEADINGS))) {
+    static QCString behaviour = Config_getEnumAsString(HEADING_AUTO_IDENTIFIER);
     QCString id;
-    id.sprintf("autotoc_md%d",autoId++);
+    if (behaviour == "DOXYGEN")
```

Comment:
Don't test on the string value but on the enum value like is done e.g. for `WARN_AS_ERROR` in `message.cpp`.
Makes it also possible to use a `switch` statement with additional benefit that some compilers will warn when a value is not handled.